samples/javascript_nodejs/84.bot-authentication-certificate/index.js

```diff
@@ -0,0 +1,119 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+const path = require('path');
+// const fs = require('fs');
+const dotenv = require('dotenv');
+const restify = require('restify');
+const { DefaultAzureCredential } = require('@azure/identity');
+const { SecretClient } = require('@azure/keyvault-secrets');
+
+// Import required bot configuration.
+const ENV_FILE = path.join(__dirname, '.env');
+dotenv.config({ path: ENV_FILE });
+
+// Import required bot services.
+// See https://aka.ms/bot-services to learn more about the different parts of a bot.
+const {
+    CloudAdapter,
+    ConfigurationBotFrameworkAuthentication
+} = require('botbuilder');
+
+// This bot's main dialog.
+const { AuthBot } = require('./authBot');
+
+(async () => {
+    try {
+        // Create HTTP server
+        const server = restify.createServer();
+        server.use(restify.plugins.bodyParser());
+
+        server.listen(process.env.port || process.env.PORT || 3978, () => {
+            console.log(`\n${ server.name } listening to ${ server.url }`);
+            console.log('\nGet Bot Framework Emulator: https://aka.ms/botframework-emulator');
+            console.log('\nTo talk to your bot, open the emulator select "Open Bot"');
+        });
+
+        // ---- Authenticate using local certificate
+        // const key = fs.readFileSync('{KeyPath}.pem', 'utf8');
+
+        // const botFrameworkAuthentication = new ConfigurationBotFrameworkAuthentication({
+        //     MicrosoftAppId: process.env.MicrosoftAppId,
+        //     CertificateThumbprint: process.env.Thumbprint,
+        //     CertificatePrivateKey: key,
+        // });
+
+        // ---- Authenticate using key vault to obtain the certificate values
```

Comment:
Okay, we have different perspectives, I had left it at the beginning so that users could see at first glance that there is another option, but I'll move it.